Sources/ProjectDescription/CompatibleXcodeVersions.swift

```diff
@@ -5,55 +5,76 @@ public enum CompatibleXcodeVersions: ExpressibleByArrayLiteral, ExpressibleByStr
     /// The project supports all Xcode versions.
     case all
 
+    case exact(Version)
+
+    case upToNextMajor(Version)
+
+    case upToNextMinor(Version)
+
     /// List of versions that are supported by the project.
-    case list([String])
+    case list([CompatibleXcodeVersions])
 
     // MARK: - ExpressibleByArrayLiteral
 
-    public init(arrayLiteral elements: [String]) {
-        self = .list(elements)
+    public init(arrayLiteral elements: [CompatibleXcodeVersions]) {
+        self = .list(elements.map { $0 })
```

Comment:
Any reason to not be just
```suggestion
        self = .list(elements)
```
?